Profile MHV and DS Logon cred retirement messaging

adamwhitlock1 opened the pull request:
## Summary

- Adds alerts around credential retirement for LOA1 users and MHV/DS Logon users when viewing the account security page.
- Uses feature toggle `profile_show_credential_retirement_messaging` and there is a corresponding BE PR to add this toggle


## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/73357

## Testing done

- New UI component has unit test coverage at 100%
- E2E tests created for contact info and account security alert cases

## Screenshots

No before screenshots since these UI components didn't exist before.

**NOTE:** the original ticket/design shows an alert showing within the contact information page, but we are not going to show the alert there, because we actually do not show a 'Sign-in email' section to DS Logon / MHV users because they don't use an email address as their sign in and instead use a username that is unique to the service.

|         | Desktop | Mobile |
| ------- | ------ | ----- |
| Account Sec LOA1  | ![Screenshot 2024-02-27 at 10 33 46 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/200b49dd-500a-4aaf-a5e5-763b209ac721) | ![Screenshot 2024-02-27 at 10 37 18 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/c527a649-9e47-4cef-9990-0f347800b0a0) |
| Account Sec DS Logon | ![Screenshot 2024-02-27 at 10 40 47 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/96e53cf5-4631-4d5c-b4a6-52a02d4e3542) | ![Screenshot 2024-02-27 at 10 41 12 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/1664842e-b67f-447b-aea2-543e8ffc48a0) |
| Account Sec MHV user | ![Screenshot 2024-02-27 at 10 45 35 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/198c7072-dbf4-4c2f-80ba-d49e42d7d86b) | ![Screenshot 2024-02-27 at 10 45 21 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/9a2725a6-b1b6-47f0-91ae-0d8b5f236d07) |



## What areas of the site does it impact?

Profile -> Account Security

## Acceptance criteria

- UI created and toggle-able via new feature flag name for specific user types (LOA1, MHV, and DS Logon)

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Screenshot of the developed feature is added

Commit:
feat: add cred alerts, selector, and toggle

Commit:
feat: alerts and tests for cred retirement

Commit:
test: e2e tests for cred retirement alerts

Commit:
Merge branch 'main' into Profile/aw/73357/mhv-ds-cred-retirement-messaging

allisonlu reviewed (commented):
good work!! lookin great 🎉  

a few questions:
1. are we implementing the alert without the dismissible option? (i'm comparing it to the Figma mockups)
2. LOA1 users - is the "Learn to create an account link" left out intentionally?
3. LOA1 users - should we hardcode MHV or DS logon in the alert message for LOA1 users? the mock user currently has `serviceName` set to `'idme'` so that's showing up in the alert, but i'm not sure if this will happen in prod.

![Screenshot 2024-02-27 at 2 52 25 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/8542413/e4741e8a-2808-46ee-a945-fab6797f5085)

adamwhitlock1 commented:
> good work!! lookin great 🎉
> 
> a few questions:
> 
> 1. are we implementing the alert without the dismissible option? (i'm comparing it to the Figma mockups)
> 2. LOA1 users - is the "Learn to create an account link" left out intentionally?
> 3. LOA1 users - should we hardcode MHV or DS logon in the alert message for LOA1 users? the mock user currently has `serviceName` set to `'idme'` so that's showing up in the alert, but i'm not sure if this will happen in prod.
> 
> ![Screenshot 2024-02-27 at 2 52 25 PM](https://private-user-images.githubusercontent.com/8542413/308319088-e4741e8a-2808-46ee-a945-fab6797f5085.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkwNjQxNzksIm5iZiI6MTcwOTA2Mzg3OSwicGF0aCI6Ii84NTQyNDEzLzMwODMxOTA4OC1lNDc0MWU4YS0yODA4LTQ2ZWUtYTk0NS1mYWI2Nzk3ZjUwODUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjdUMTk1NzU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTg4NjE2YjRiYjBkYWIwMDJmZWQzNGEyZTAwNWE5NTA0MTcyMzMyMDhjNWJiN2EzMzE5ZWRjYjI3OTk5NzE0ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.mMxJaPYbER4S55XY8vHJfM5nJzVU6TOwzy5kRXEBBD4)

You are so right... I had made a mental note to include session storage to make the alert dismissible.

Not sure how I missed the link in there 🤦🏼 thanks for catching that.

I could make a mock user that is MHV and DS Logon... that is probably a good idea just to have them available

Commit:
add mock loa1 users, session storage dismiss alert, change display behavior of loa1 alert

Commit:
test: add unit tests for acct sec cont and update alert test

adamwhitlock1 commented:
@allisonlu thank you for pointing out those items.

After talking to Liz, we went the route of only showing that LOA1 alert when the user in MHV/DS Logon, and showing the older alert for Id.me users.

- updated that component logic in the `AccountSecuirtyContent` component, added unit tests for that component in general since it had none, and also tested what text shows for those cases.
- added useSession storage for the dismissible alert version and made it actually dismissible now.
- lastly I added mock users to the api for loa1 mhv and dslogon account types for local testing and such

Commit:
Merge branch 'main' into Profile/aw/73357/mhv-ds-cred-retirement-messaging

Commit:
test: update e2e for alert on account security

adamwhitlock1 commented:
@allisonlu, @DanielMiller-agile6, @micahchiang  would yall mind re-reviewing this when you have a chance?

I fixed the items that were previously mentioned, and it should be good to go now

allisonlu reviewed (approved):
looks great adam!! 🎉

Andrew565 commented on src/applications/personalization/profile/components/alerts/CredentialRetirementAlerts.jsx:
```diff
@@ -0,0 +1,95 @@
+        <div>
+          <p>
+            Before we give you access to your VA.gov profile, we need to make
+            sure you’re you—and not someone pretending to be you. This helps us
```
Needs space around the hyphen. "you - and"

adamwhitlock1 replied:
Pulled this content directly from Figma so it does reflect what is in the design at this time. I will follow up with designer and see what they say. Thanks @Andrew565 for reviewing!
